include/language.php

```diff
@@ -490,4 +693,288 @@ public function set_url_scheme() {
+		 *         language: array{
+		 *             term_id: positive-int,
+		 *             term_taxonomy_id: positive-int,
+		 *             count: int<0, max>
+		 *         },
+		 *         term_language: array{
+		 *             term_id: positive-int,
+		 *             term_taxonomy_id: positive-int,
+		 *             count: int<0, max>
+		 *         }
+		 *     },
+		 *     name: non-empty-string,
+		 *     slug: non-empty-string,
+		 *     locale: non-empty-string,
+		 *     w3c: non-empty-string,
+		 *     flag_code: non-empty-string,
+		 *     term_group: int,
+		 *     is_rtl: int<0, 1>,
+		 *     mo_id?: positive-int,
+		 *     facebook?: non-empty-string,
+		 *     home_url?: non-empty-string,
+		 *     search_url?: non-empty-string,
+		 *     host?: non-empty-string,
+		 *     flag_url?: non-empty-string,
+		 *     flag?: non-empty-string,
+		 *     custom_flag_url?: non-empty-string,
+		 *     custom_flag?: non-empty-string,
+		 *     page_on_front?:positive-int,
+		 *     page_for_posts?:positive-int
+		 * }
+		 */
+		return array_intersect_key(
+			$data,
+			array(
+				'term_props'      => null,
+				'name'            => null,
+				'slug'            => null,
+				'locale'          => null,
+				'w3c'             => null,
+				'flag_code'       => null,
+				'term_group'      => null,
+				'is_rtl'          => null,
+				'mo_id'           => null,
+				'facebook'        => null,
+				'home_url'        => null,
+				'search_url'      => null,
+				'host'            => null,
+				'flag_url'        => null,
+				'flag'            => null,
+				'custom_flag_url' => null,
+				'custom_flag'     => null,
+				'page_on_front'   => null,
+				'page_for_posts'  => null,
+			)
+		);
+	}
+
+	/**
+	 * Validates and sanitizes a `term_prop` entry.
+	 * This doesn't verify that the term exists.
+	 *
+	 * @since 3.4
+	 *
+	 * @param array $data Data to process.
+	 * @return int[]|null
+	 *
+	 * @phpstan-return array{
+	 *     term_id: positive-int,
+	 *     term_taxonomy_id: positive-int,
+	 *     count: int<0, max>
+	 * }|null
+	 */
+	private static function validate_term_prop_group( array $data ) {
+		if ( empty( $data['term_id'] ) ) {
+			return null;
+		}
+
+		if ( ! is_numeric( $data['term_id'] ) || $data['term_id'] < 1 ) {
+			return null;
+		}
+
+		$data['term_id'] = abs( (int) $data['term_id'] );
+
+		if ( empty( $data['term_taxonomy_id'] ) ) {
+			return null;
+		}
+
+		if ( ! is_numeric( $data['term_taxonomy_id'] ) || $data['term_taxonomy_id'] < 1 ) {
+			return null;
+		}
+
+		$data['term_taxonomy_id'] = abs( (int) $data['term_taxonomy_id'] );
+
+		if ( empty( $data['count'] ) ) {
+			$data['count'] = 0;
+		} elseif ( $data['count'] < 0 ) {
+			$data['count'] = 0;
+		} else {
+			$data['count'] = abs( (int) $data['count'] );
+		}
+
+		/**
+		 * @phpstan-var array{
+		 *     term_id: positive-int,
+		 *     term_taxonomy_id: positive-int,
+		 *     count: int<0, max>
+		 * }
+		 */
+		return array_intersect_key(
```

Comment:
What happens if some new language term requires a new prop?
Should we add a filter to allow to sanitize it?

Reply:
We should keep it hard-coded so far imho: if we allow other properties, we would also need to do something about sanitization and validation.

Reply:
Fine, I understand your point.